### PR TITLE
feat: Add CRCR RH callback 

### PR DESCRIPTION
**Pull request by kmehant**

### CODE REVIEW

**Review by ashokponkumar — DISMISSED** [review]

File: .github/workflows/build_test_pytorch_source.yaml [hunk 1]
```diff
@@ -407,3 +407,76 @@ jobs:
...
+          WORKFLOW_NAME: ${{ github.workflow }}
+          # Same convention as the pytorch/pytorch CRCR relay's delivery-id
+          # above -- keyed on the bare resolved pytorch_sha, not a composite
+          # string.
+          DELIVERY_ID: ${{ inputs.pytorch_sha }}
+          # Same rule the CRCR relay callbacks above use: only a real
+          # "success" once tests actually ran (total > 0).
+          CONCLUSION: >-
+            ${{
+              (fromJSON(steps.test_results.outputs.test_results_json).total > 0) && 'success' || 'failure'
```
Comment (subinz1) [inline .github/workflows/build_test_pytorch_source.yaml]: This checks whether tests *ran* (`total > 0`) but not whether they *passed*. If 100 tests execute and 50 fail, `total > 0` is still true, so this reports `success` even though the run actually failed.

Consider also checking failures are zero:

```yaml
CONCLUSION: >-
  ${{ 
    fromJSON(steps.test_results.outputs.test_results_json).total > 0
    && fromJSON(steps.test_results.outputs.test_results_json).failed == 0
    && 'success' || 'failure'
  }}
```

Or simply derive from job status (which already reflects prior step failures):

```yaml
CONCLUSION: ${{ job.status }}
```

Comment (kmehant) [reply .github/workflows/build_test_pytorch_source.yaml]: @subinz1 Have updated it!

File: .github/workflows/build_test_pytorch_source.yaml [hunk 2]
```diff
@@ -427,3 +427,78 @@ jobs:
...
+          RUN_ATTEMPT: ${{ github.run_attempt }}
+          WORKFLOW_NAME: ${{ github.workflow }}
+          # Same convention as the pytorch/pytorch CRCR relay's delivery-id
+          # above -- keyed on the bare resolved pytorch_sha, not a composite
+          # string.
+          DELIVERY_ID: ${{ inputs.pytorch_sha }}
+          # Same rule the CRCR relay callbacks above use: only a real
+          # "success" once tests actually ran (total > 0).
+          CONCLUSION: >-
+            ${{
```
Comment (subinz1) [inline .github/workflows/build_test_pytorch_source.yaml]: CONCLUSION fix looks good — `failed == 0` check addresses the previous concern.

One edge case: if the build fails before tests, `steps.test_results.outputs.test_results_json` may be empty/undefined. `fromJSON()` on an empty string errors in the expression evaluator, and since this is in the `env:` block (evaluated before the step runs), `CONCLUSION` could end up as `""` — an invalid value for the relay.

Guard the `fromJSON` call:

```yaml
CONCLUSION: >-
  ${{
    steps.test_results.outputs.test_results_json
    && fromJSON(steps.test_results.outputs.test_results_json).total > 0
    && fromJSON(steps.test_results.outputs.test_results_json).failed == 0
    && 'success' || 'failure'
  }}
```

The leading `steps.test_results.outputs.test_results_json &&` short-circuits to `'failure'` when the output is empty, instead of erroring on `fromJSON()`.

Comment (kmehant) [reply .github/workflows/build_test_pytorch_source.yaml]: Done!

**Review by subinz1 — APPROVED** [review]
LGTM!

**Review by ashokponkumar — APPROVED** [review]
